client/app.jsx

```diff
@@ -21,13 +25,25 @@ export default class App extends React.Component {
     });
   }
 
+  getSearchResults(search) {
+    this.setState({ searchTerm: search });
+    fetch('https://api.seatgeek.com/2/events?q=' + search + '&per_page=50&client_id=OTEzNzY5NnwxNjM1Nzk3ODUzLjE2OTAyNTI')
+      .then(request => request.json())
+      .then(data => {
+        this.setState({
+          searchResults: data.events,
+          meta: data.meta
+        });
```

Comment:
Here is where you want to put the route change, but you want it to happen after the state finishes updating. So we will use a second argument for setState here, a callback function that gets called once state has finished updating. Once you fix this and remove the anchor tag from the search bar it might fix the warning you were getting. Here is what it will look like:

```JAVASCRIPT
this.setState({
  searchResults: data.events,
  meta: data.meta
}, () => {
  location.hash = '#results';
});
```